Realm/Realm.Weaver/Analytics/AnalyticsUtils.cs

```diff
@@ -0,0 +1,164 @@
+﻿// ////////////////////////////////////////////////////////////////////////////
+// //
+// // Copyright 2022 Realm Inc.
+// //
+// // Licensed under the Apache License, Version 2.0 (the "License")
+// // you may not use this file except in compliance with the License.
+// // You may obtain a copy of the License at
+// //
+// // http://www.apache.org/licenses/LICENSE-2.0
+// //
+// // Unless required by applicable law or agreed to in writing, software
+// // distributed under the License is distributed on an "AS IS" BASIS,
+// // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// // See the License for the specific language governing permissions and
+// // limitations under the License.
+// //
+// ////////////////////////////////////////////////////////////////////////////
+
+using System;
+using System.Linq;
+using System.Net.NetworkInformation;
+using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
+using System.Text.RegularExpressions;
+using Mono.Cecil;
+using static RealmWeaver.Metric;
+
+namespace RealmWeaver
+{
+    internal static class AnalyticsUtils
+    {
+        // TODO andrea: https://github.com/realm/realm-dotnet/issues/2706#event-7877818404
+        // add a stable ID not reliant on MACAddress
+        public static string AnonymizedUserID
```

Comment:
Why is this a property while the others are methods?

Reply:
I just got what we were doing in the previous work. Knowing that I have to work on the github issue, I did not put effort on changing it for now.

Reply:
Reasonable!